main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,113 @@
+#include "main/lsp/requests/go_to_implementation.h"
+#include "core/lsp/QueryResponse.h"
+#include "main/lsp/json_types.h"
+#include "main/lsp/lsp.h"
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+
+vector<core::SymbolRef> findChildren(const core::GlobalState &gs, core::SymbolRef ancestorClass) {
```

Comment:
Nit: You should put these freestanding non-class functions into an anonymous namespace like so:

```
namespace {
  // functions defined here
}
```

The anonymous namespace tells the compiler that these functions will be compiled with "internal linkage", meaning that they cannot be referenced outside of this `.cc` file (by default, this is not the case!).

Reply:
btw, there's a new function in `rename.cc` that may be useful called `getSubclasses`:

https://github.com/sorbet/sorbet/blob/HEAD/main/lsp/requests/rename.cc#L61

It's a bit of a misnomer; it finds all subclasses and, if a module, all modules that have the module mixed in. So, it exactly matches your use case here. It's much more efficient than your current loop, as it reimplements `derivesFrom` in a manner that memoizes intermediate results.

I'd be down to move it to a helper method on `LSPTask` so it can be reused in a place like this.

Reply:
You don't have to change your code immediately to use this method or anything; that can be a later change.

Reply:
I moved this code to `Symbols.cc`, because it didn't fit well into `LSPTask`. I also added a param with a default value to filter the self out of the result.